### PR TITLE
chore(app): move initialization logic outside of the app and into keepers package

### PR DESCRIPTION
**Pull request by zale144**
## Description

In the PR #972 the upgrade handlers are refactored so that they use an interface for the upgrades themselves and keepers are extracted to be in a separate package. 

Here we go a step further and move the initializing logic out of `app.go` and into the new package `keepers`.

----

Closes [#320](https://github.com/dymensionxyz/research/issues/320)

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----;

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

**Comment by omritoptix** [comment]
@zale144 please give it a meaningful pr title. thanks.

File: app/keepers/keys.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+func (*AppKeepers) ModuleAccountAddrs() map[string]bool {
+	modAccAddrs := make(map[string]bool)
+	for acc := range maccPerms {
+		modAccAddrs[authtypes.NewModuleAddress(acc).String()] = true
+	}
+
+	// exclude the streamer as we want him to be able to get external incentives
+	modAccAddrs[authtypes.NewModuleAddress(streamermoduletypes.ModuleName).String()] = false
+	modAccAddrs[authtypes.NewModuleAddress(txfeestypes.ModuleName).String()] = false
+	return modAccAddrs
```
Comment (danwt) [inline app/keepers/keys.go:103]: this is not really anything keys related

File: app/keepers/keepers.go [hunk 2]
```diff
@@ -79,4 +137,442 @@ type AppKeepers struct {
...
 	DenomMetadataKeeper *denommetadatamodulekeeper.Keeper
+
+	// keys to access the substores
+	keys    map[string]*storetypes.KVStoreKey
+	tkeys   map[string]*storetypes.TransientStoreKey
+	memKeys map[string]*storetypes.MemoryStoreKey
+}
+
+func (a *AppKeepers) InitNormalKeepers(
+	appCodec codec.Codec,
```
Comment (danwt) [inline app/keepers/keepers.go:148]: needs docstring
I dont know what a normal keeper is

**Review by danwt — APPROVED** [review]
nice refactor
I understand the keeprs pckage now

File: app/keepers/keepers.go [hunk 3]
```diff
@@ -76,7 +134,442 @@ type AppKeepers struct {
...
+
+	// Create static IBC router, add transfer route, then set and seal it
+	ibcRouter := ibcporttypes.NewRouter()
+	ibcRouter.AddRoute(ibctransfertypes.ModuleName, a.TransferStack)
+	a.IBCKeeper.SetRouter(ibcRouter)
+}
+
+func (a *AppKeepers) SetupHooks() {
+	// register the staking hooks
+	a.StakingKeeper = *a.StakingKeeper.SetHooks(
```
Comment (mtsitrin) [inline app/keepers/keepers.go:481]: don't u overwrite here the `a.StakingKeeper` while already pass `&a.StakingKeeper` to different `.NewKeeper()` methods?

**Pull request merged**